scripts/drop_all_open_ossrh_repositories.py

```diff
@@ -0,0 +1,133 @@
+import os
+import sys
+import requests
+import urllib.parse
+
+# --- Configuration ---
+SONATYPE_API_BASE_URL = "https://ossrh-staging-api.central.sonatype.com/manual"
+SEARCH_REPOSITORIES_URL = f"{SONATYPE_API_BASE_URL}/search/repositories"
+DROP_REPOSITORY_BASE_URL = f"{SONATYPE_API_BASE_URL}/drop/repository" # Key will be appended
+TIMEOUT_SECONDS = 60
+
+def _get_auth_credentials():
+    # Retrieves Sonatype username and password from environment variables.
+    # Exits with an error if they are not set.
+    username = os.environ.get("SONATYPE_CENTRAL_PORTAL_USERNAME")
+    password = os.environ.get("SONATYPE_CENTRAL_PORTAL_PASSWORD")
+
+    if not username or not password:
+        print("Error: SONATYPE_CENTRAL_PORTAL_USERNAME and SONATYPE_CENTRAL_PORTAL_PASSWORD environment variables must be set.")
+        sys.exit(1)
+    return username, password
+
+def _make_api_request(method, url, auth, params=None, headers=None, expected_success_codes=None):
+    # Makes an HTTP request and handles common errors.
+    # Returns the response object on success, None on failure or request exception.
+    if headers is None:
+        headers = {}
+    headers.setdefault("Accept", "application/json") # Default Accept header
+
+    if expected_success_codes is None:
+        expected_success_codes = [200]
+
+    print(f"Making {method} request to: {url} with params: {params if params else 'None'}")
+    try:
+        response = requests.request(method, url, auth=auth, params=params, headers=headers, timeout=TIMEOUT_SECONDS)
```

Comment:
Nit: Would be nice and consistent if we wrote argument names for `method` and `url` too.